lib/ex_aws/s3.ex

```diff
@@ -488,22 +489,54 @@ defmodule ExAws.S3 do
 
   Handles initialization, uploading parts concurrently, and multipart upload completion.
 
-  ## Examples
+  ## Uploading a stream
+
+  Streams that emit binaries may be uploaded directly to S3. Each binary will be uploaded
+  as a chunk, so it must be at least 5 megabytes in size. The `S3.Upload.stream_file`
+  helper takes care of reading the file in 5 megabyte chunks.
   ```
   "path/to/big/file"
-  |> S3.Upload.stream_file!
+  |> S3.Upload.stream_file
   |> S3.upload("my-bucket", "path/on/s3")
-  |> ExAws.request! #=> {:ok, :done}
+  |> ExAws.request! #=> :done
   ```
 
-  ## Notes
+  ## Uploading a flow
+
+  GenStage Flows can also be uploaded directly to S3. However, due to the concurrent
+  nature of flows, the producer must emit events in the following format
+  `{binary, index}` to ensure they are uploaded in the correct order. Binaries
+  will be uploaded as a chunk and must be at least 5 megabytes in size.
+  ```
+  enumerable = ["hello world"] |> Enum.with_index(1)
```

Comment:
We probably want to promote `Stream.with_index` here as most people's enumerables are gonna be lazy. If it isn't lazy we're gonna lose a lot of the memory savings this approach offers.